### ISSUE TITLE
The problem with tags... (question really)

### ISSUE THREAD

**Issue by memelet** [awailly/cis-ubuntu-ansible#99]
I have certainly playbooks that need (ok want) to skip some checks. For example, 12.8 which checks for un-owned files when I am rebuilding an elasticsearch node and reattaching an existing EBS volume. When CIS runs, the elasticsearch user has not been created yet and the elasticsearch role that will set the owner of the data files on the EBS has not run yet. So the gazillion elasticsearch files on the EBS are not yet owned, which generates a gazillion errors.

What I really want to say -- in that specific playbook -- don't run 12.8. But I can't because the only way to turn off bits of CIS is via tags. And tags can only be set on the command line. 

So what do others do? Do you just ignore that you get a 10k errors when provisioning a new node. Or do you always the run the CIS role last in the playbook. (I have considered this, but I want to create a base image with CIS configured since it takes so long.)

**Comment by awailly** [comment]
On my side I always run the CIS playbook at the end as it check that everything is correct. I also feel like running it before open the door for the next playbook to destroy everything the CIS playbook was trying to achieve. This way, I can launch the CIS benchmark several times, say once a week, and be sure that my node is fully compliant.

> (I have considered this, but I want to create a base image with CIS configured since it takes so long.)

I miss why you can't have the base image and the CIS playbook in the last position at the same time. Can you clarify this point? I will be happy to investigate how we can improve performances with your use case.

**Comment by pchaigno** [comment]
> What I really want to say -- in that specific playbook -- don't run 12.8. But I can't because the only way to turn off bits of CIS is via tags. And tags can only be set on the command line.

@memelet Just to clarify, you'd like a way to use cis-ansible-ubuntu in your own playbook and skip some of its tasks, is that it?

> I miss why you can't have the base image and the CIS playbook in the last position at the same time.

@awailly I think he'd like to create a hardened base image on which he can build more specific images (@memelet correct me if I'm mistaken).
What's not clear to me is why part of the Elasticsearch installation is performed in the base image and part of it after... In other words, why can't the Elasticsearch user be defined in the base image? (I'm unfamiliar with Elasticsearch, so I may be missing something obvious here :/)

**Comment by memelet** [comment]
@awailly, @pchaigno 

> I miss why you can't have the base image and the CIS playbook in the last position at the same time. Can you clarify this point? I will be happy to investigate how we can improve performances with your use case.

Yes, I meant as a hardened base image from which instances are bootstrapped using ansible. But given @awailly's reasoning for running last I may forgo this desire of a _base_ image.

@pchaigno If I created a base image it should be closed to whether the ultimate instance was for elasticsearch, kafka, mesos, etc. So I would not want to do role specific stuff in the base image.

I tried creating the user with a fixed UID/GID, but this has problems when running the playbook on a cluster of instances where some are already running and others that have just been created. On the former, the user creation fails because the user is associated with active process. 

So I have taken a different approach. In playbooks/roles that configure an application (eg elasticsearch) on an instance that may have "taken over" an EBS volume from a previous lost instance, that playbook is responsible for ensuring the correct ownership of all the data that is on that EBS volume before its service is started. 

Then I have factored out CIS from my base playbook and have moved to last in the including playbook. So far this is working quite well. And besides, I really like idea of ensuring that previous roles/plays have not violated CIS. (Well, I also run OSSEC on every node which does periodic audits of CIS, so do find out of something later has caused havoc.)

Given the above ramble, I think I no longer want to skip bits of the CIS role, so the issue with tags is now a non-issue for me. So I thank you both for the feedback and will close this.
